gap/subprograms.g

```diff
@@ -879,20 +890,26 @@ GAUSS_GlueR := function(rank, ncols, galoisField, nrows, D, R, a, b)
                  fi;
                  continue;
              fi;
-             C{[tmpR .. tmpR + DimensionsMat(R[i][j])[1]-1 ]}{[tmpC .. tmpC + DimensionsMat(R[i][j])[2]-1 ]} := R[i][j];
+             C{[tmpR .. tmpR + DimensionsMat(R[i][j])[1]-1 ]}
+                {[tmpC .. tmpC + DimensionsMat(R[i][j])[2]-1 ]}
+                := R[i][j];
             tmpC := tmpC + DimensionsMat(R[i][j])[2];
         od;
         tmpR := tmpR + rows[i];
     od;
 
-    C := TransposedMat( GAUSS_RRF( galoisField,TransposedMat(C), -IdentityMat( rank,galoisField ),w  ) );
-
+    idMat := IdentityMat( rank,galoisField );
+    ConvertToMatrixRepNC(idMat, galoisField);
+    C := TransposedMat( GAUSS_RRF( galoisField, TransposedMat(C), -idMat, w ) );
+    # FIXME Is it safe to call ConvertToMatrixRepNC(C, galoisField) here?
+    # ConvertToMatrixRepNC(C, galoisField);
     return rec( C := C, w := w );
 end;
 
 GAUSS_GlueK := function(v, rank, galoisField, a, b, E, K, mat)
-    local D, X, rows, i, j, tmpR, tmpC, tmp;
+    local D, X, rows, i, j, tmpR, tmpC, tmp, nullMat;
 
+    # FIXME convert both?
```

Comment:
Note to myself:
We can't convert `D` here because `D{list1}{list2} := ..` can't be done for compressed matrices. See #187.